src/components/SlateEditor/plugins/details/onBackspace.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2016-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Block, Document, Editor, Inline, Node, Point } from 'slate';
+
+type ParentNode = Document | Block | Inline | null;
+
+const shouldDeleteIfInsideOfDetailsblock = (
+  blockNode: Node,
+  editor: Editor,
+  node: Node,
+  paragraphNode: ParentNode,
+  previousNode: ParentNode,
+  start: Point,
+): boolean => {
+  const document = editor.value.document;
+  const previousText = document.getPreviousText(node.key);
+  if (previousText === null) return false;
+  const previousTextParent = document.getParent(previousText.key);
+  if (previousTextParent === null) return false;
+  const previousTextGrandparent = document.getParent(previousTextParent.key);
+  const isInTheMiddleOfText = start.offset < node.text.length;
+
+  return (
+    // 1. checks if the previous textline is part of the same component
+    (blockNode.text.includes(previousText.text) || isInTheMiddleOfText) &&
+    // 2. checks if the previous textline shares the same grandparent
+    (blockNode === previousTextGrandparent || isInTheMiddleOfText) &&
+    // 3. prevent delete with backspace if the cursor is at the start of the text solution/details element a.k.a
+    // if the is nothing to delete in the current element.
+    // we are preventing the deletion of non-text elements
+    (start.offset !== 0 ||
+      previousNode?.type === 'paragraph' ||
+      previousNode?.type === 'br') &&
```

Comment:
Dette gir ikkje heilt meining i details-plugin. Det er kun dersom teksten er tom og summary er tom at denne skal fjerne boksen.

Reply:
Den dagen vi hadde retrospektiv så spurte jeg i plenum om hva tanken er om sletting av slate elementer. Vi kom fram til at slette knappen skulle brukes for å slette slate elementer og forhindre sletting ved delete/backspace. Så jeg tok utgangspunktet i det når jeg lagde koden. Tanken var at når du står i en tom p-tag inn i details og det er ingen tekst i summary så skal ikke summary eller details boksen slettes med backspace. `PreviousNode` blir i det tilfelle `summary` og sjekken returnerer false. Grunnen til at det sjekkes på p-tag eller br er at det er kun tekst som skal slettes, dersom det er noe tekst før eller etter markøren (p-tags) skal da backspace kunne slette. Det var hele tanken. 

Nå som du har jobbet selv med slate har du kanskje noe input på om det skal gjøres slik?